From a715480416b33b0bacd2b58ec42b9c64bdb21c0c Mon Sep 17 00:00:00 2001
From: Michael Weiser <michael.weiser@gmx.de>
Date: Fri, 19 Aug 2016 10:24:40 +0100
Subject: [PATCH 091/104] Fix test compile time warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

A number of tests cause compiler warnings like this:

hashcrypt_speed.c: In function ‘hash_data’:
hashcrypt_speed.c:101:2: warning: implicit declaration of function ‘alarm’ [-Wimplicit-function-declaration]
  alarm(5);
  ^~~~~
hashcrypt_speed.c: In function ‘main’:
hashcrypt_speed.c:203:2: warning: implicit declaration of function ‘close’ [-Wimplicit-function-declaration]
  close(fdc);
  ^~~~~

Fix by including unistd.h.
---
 tests/hashcrypt_speed.c | 1 +
 tests/sha_speed.c       | 1 +
 tests/speed.c           | 1 +
 3 files changed, 3 insertions(+)

diff --git a/tests/hashcrypt_speed.c b/tests/hashcrypt_speed.c
index 045bf8e..10c9f00 100644
--- a/tests/hashcrypt_speed.c
+++ b/tests/hashcrypt_speed.c
@@ -21,6 +21,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <sys/types.h>
diff --git a/tests/sha_speed.c b/tests/sha_speed.c
index 9f2c8cc..30b40f5 100644
--- a/tests/sha_speed.c
+++ b/tests/sha_speed.c
@@ -21,6 +21,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <sys/types.h>
diff --git a/tests/speed.c b/tests/speed.c
index 3b36db1..fc38a63 100644
--- a/tests/speed.c
+++ b/tests/speed.c
@@ -22,6 +22,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <sys/types.h>
-- 
2.10.2

